benchmarks/opencl/caslock-sc.cl

```diff
@@ -0,0 +1,15 @@
+static void lock(global uint* l) {
+    while (atom_cmpxchg(l, 0, 1) == 1) {}
+}
+
+static void unlock(global uint* l) {
+    atom_xchg(l, 0);
+}
+
+__kernel void mutex_test(global uint* l, global int* x, global int* A) {
```

Comment:
Well, you added this benchmark yourself. And I would say that it would be good to have OpenCL-1.0 version for the other locks for consistency. Anyway, added storing output value to the array, so it is not useless anymore.